### PR TITLE
Add additional thoughts on rolling out breaking changes

### PR DESCRIPTION
**Pull request by kfcampbell**
Today we talked about how to roll out breaking changes safely in the octokit.js ecosystem. This PR captures that discussion in a more durable place.

### CODE REVIEW

**Comment by wolfy1339** [comment]
It may be good to note that for changes requiring type changes, `@octokit/types` needs to be updated

File: MAINTAINING.md [hunk 1]
```diff
@@ -21,6 +21,22 @@ If the pull request looks good but does not follow the commit conventions, use t
...
+1. plugins
+1. auth strategies
+1. core
+1. \*-methods
+1. oauth-app
+1. webhooks
+1. app
+1. octokit/octokit.js
+1. octokit/rest.js
+
```
Comment (kfcampbell) [inline MAINTAINING.md:42]: Do we (or should we) have a procedure for how often we diff the `beta` branch with `main` and create new releases?

Comment (gr2m) [reply MAINTAINING.md:42]: Usually the two shouldn't diverge, `beta` branches should be short lived. Or is your question about something else?

**Review by gr2m — APPROVED** [review]

File: MAINTAINING.md (hunk 1 above)
Comment (wolfy1339) [reply MAINTAINING.md:42]: I don't think there is a procedure, however one should create a draft PR for the next major version merging `beta` into `main`, ex: https://github.com/octokit/request.js/pull/586 and then mark it ready for review when it is ready

Comment (kfcampbell) [reply MAINTAINING.md:42]: In that case, should I amend the procedure to contain something like: 

"After the breaking changes are combined and tested in the `beta` branch, merge the `beta` branch into `main` using a PR with "BREAKING CHANGE" in the body as documented above. `beta` branches are intended to be short-lived."

**Comment by kfcampbell** [comment]
Good idea @wolfy1339! Would that go at the top of the chain since it's a dependency of many other modules?

**Comment by wolfy1339** [comment]
That sounds right

File: MAINTAINING.md (hunk 1 above)
Comment (kfcampbell) [reply MAINTAINING.md:42]: I'ved added similar language to the PR; please let me know what you think!

**Comment by kfcampbell** [comment]
I've updated the list accordingly!

File: MAINTAINING.md (hunk 1 above)
Comment (gr2m) [reply MAINTAINING.md:42]: Good pony by @wolfy1339, we should create beta pull requests as drafts until all planned breaking changes have been merged into it

Comment (gr2m) [reply MAINTAINING.md:42]: Maybe add that

1. beta branches should be deleted first and re-created based on main
2. Once the first pull request is merged into beta, create a draft pull request from beta to main with the title vX where X is the next breaking version.
3 Once all breaking changes are merged into beta, test and set the pull request to ready for review

**Review by nickfloyd — APPROVED** [review]

File: MAINTAINING.md (hunk 1 above)
Comment (kfcampbell) [reply MAINTAINING.md:42]: Sounds good! I've now done so.

**Pull request merged**